src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -159,13 +183,53 @@ public static String createStreamEvent(final Cursor cursor, final List<String> e
 
     private void sendBatch(final NakadiCursor topicPosition, final List<String> currentBatch)
             throws IOException {
-        // create stream event batch for current partition and send it; if there were
-        // no events, it will be just a keep-alive
-        final String streamEvent = createStreamEvent(cursorConverter.convert(topicPosition), currentBatch);
-        final byte[] batchBytes = streamEvent.getBytes(UTF8);
-        outputStream.write(batchBytes);
-        bytesFlushedMeter.mark(batchBytes.length);
-        outputStream.flush();
+        if(featureToggleService.isFeatureEnabled(FeatureToggleService.Feature.SEND_BATCH_VIA_OUTPUT_STREAM)) {
+            writeStreamEvent(outputStream, cursorConverter.convert(topicPosition), currentBatch);
+        } else {
+            // create stream event batch for current partition and send it; if there were
+            // no events, it will be just a keep-alive
+            final String streamEvent = createStreamEvent(cursorConverter.convert(topicPosition), currentBatch);
+            final byte[] batchBytes = streamEvent.getBytes(UTF8);
+            outputStream.write(batchBytes);
+            bytesFlushedMeter.mark(batchBytes.length);
+            outputStream.flush();
+        }
+    }
+
+    @VisibleForTesting
+    void writeStreamEvent(final OutputStream os, final Cursor cursor, final List<String> events)
```

Comment:
Also I think it will make sense to move the low-level writing together with bytes constants into a separate class.